### PR TITLE
Update Syntax.md

### PR DESCRIPTION
**Pull request by mooskagh**
"Stored by value" is more correct way to explain that example.

Mutable vs immutable means that you cannot mutate the value (e.g. list vs tuple in Python), and the example shows that `var2` is actually mutable.

Copying/storing a reference vs value is what what matters in the assignment, in Python `a=b` means `a` and `b` are references to the same list, while in meson `a=b` copies the value of `b` into `a`.

### CODE REVIEW

**Comment by mooskagh** [comment]
I've read the document further and realized what exactly was meant by immutability.

I've updated it to be less confusing, but now it's a bit too wordy.

**Review by dcbaker — APPROVED** [review]

**Comment by eli-schwartz** [comment]
Should be amended and force pushed, not made into 2 commits. :/

**Comment by eli-schwartz** [comment]
Of course github is allergic to good formatting, lol, so the resulting squashed commit message is rather run-on...
